frontend/lib/src/components/elements/Popover/Popover.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement } from "react"
+
+import { useTheme } from "@emotion/react"
+import { ExpandMore, ExpandLess } from "@emotion-icons/material-outlined"
+
+import { hasLightBackgroundColor } from "@streamlit/lib/src/theme"
+import { StyledIcon } from "@streamlit/lib/src/components/shared/Icon"
+import { Block as BlockProto } from "@streamlit/lib/src/proto"
+import BaseButton, {
+  BaseButtonKind,
+  BaseButtonSize,
+} from "@streamlit/lib/src/components/shared/BaseButton"
+import StreamlitMarkdown from "@streamlit/lib/src/components/shared/StreamlitMarkdown"
+import IsSidebarContext from "@streamlit/lib/src/components/core/IsSidebarContext"
+
+import { Popover as UIPopover, TRIGGER_TYPE, PLACEMENT } from "baseui/popover"
+
+import { StyledPopoverButtonIcon } from "./styled-components"
+
+export interface PopoverProps {
+  element: BlockProto.Popover
+  empty: boolean
+}
+
+const Popover: React.FC<PopoverProps> = ({
+  element,
+  empty,
+  children,
```

Comment:
Does `children` have to be defined in props?

Reply:
I don't think so since it is not supposed to be provided as actual props. I think it's something that gets automatically injected by React. We are also not doing this in Expander, but not sure if that's the suggested way to do this.